Docs: Fix typos, improve formatting on page `Atomic` engine (#74690)
ClickHouse/ClickHouse: docs/en/engines/database-engines/atomic.md

# Atomic 

> The `Atomic` engine supports non-blocking [`DROP TABLE`](#drop-detach-table) and [`RENAME TABLE`](#rename-table) queries and atomic [`EXCHANGE TABLES`](#exchange-tables) queries. The `Atomic` database engine is used by default. 

Reviewer: ```suggestion
The `Atomic` engine supports non-blocking [`DROP TABLE`](#drop-detach-table) and [`RENAME TABLE`](#rename-table) queries, and atomic [`EXCHANGE TABLES`](#exchange-tables) queries. The `Atomic` database engine is used by default. 
```

:::note

> On ClickHouse Cloud, the `Replicated` database engine is used by default.

Reviewer: ```suggestion
By default, the ClickHouse Cloud uses the Replicated database engine.
```

:::

## Creating a Database {#creating-a-database}

```sql
CREATE DATABASE test [ENGINE = Atomic];
```

> ## Specifics and recommendations {#specifics-and-recommendations}

Reviewer: ```suggestion
## Specifics and Recommendations {#specifics-and-recommendations}
```

### Table UUID {#table-uuid}

> All tables in database `Atomic` have persistent [UUID](../../sql-reference/data-types/uuid.md) and store data in directory:

Reviewer: ```suggestion
Each table in the `Atomic` database has a persistent [UUID](../../sql-reference/data-types/uuid.md) and stores its data in the following directory:
```

```
/clickhouse_path/store/xxx/xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy/
```

Where `xxxyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy` is the UUID of the table.

> Usually, the UUID is generated automatically, but the user can also explicitly specify the UUID in the same way when creating the table, although this is not recommended. 

Reviewer: ```suggestion
By default, the UUID is generated automatically. However, users can explicitly specify the UUID when creating a table, though this is not recommended.
```

> For example:

Reviewer: Suggestion: We can change the Note to:
"You can enable the [show_table_uuid_in_table_create_query_if_not_nil](../../operations/settings/settings.md#show_table_uuid_in_table_create_query_if_not_nil) setting to include the UUID in the output of the `SHOW CREATE` query."

This is active voice making it more direct & clear.

### RENAME TABLE {#rename-table}

> [`RENAME`](../../sql-reference/statements/rename.md) queries are performed without changing the UUID or moving table data. These queries do not wait for the completion of queries using the table and are executed instantly.

Reviewer: ```suggestion
[`RENAME`](../../sql-reference/statements/rename.md) queries do not modify the UUID or move table data. These queries execute immediately and do not wait for other queries that are using the table to complete.
```

### DROP/DETACH TABLE {#drop-detach-table}

> When using `DROP TABLE`, no data is removed. The `Atomic` engine just marks the table as dropped by moving it's metadata to `/clickhouse_path/metadata_dropped/` and notifies the background thread. The delay before the final table data deletion is specified by the [`database_atomic_delay_before_drop_table_sec`](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.

Reviewer: ```suggestion
When using `DROP TABLE`, no data is removed. The `Atomic` engine just marks the table as dropped by moving its metadata to `/clickhouse_path/metadata_dropped/` and notifying the background thread. The delay before the final deletion of the table's data is controlled by the [`database_atomic_delay_before_drop_table_sec`](../../operations/server-configuration-parameters/settings.md#database_atomic_delay_before_drop_table_sec) setting.
```

> You can specify synchronous mode using `SYNC` modifier. Use the [`database_atomic_wait_for_drop_and_detach_synchronously`](../../operations/settings/settings.md#database_atomic_wait_for_drop_and_detach_synchronously) setting to do this. In this case `DROP` waits for running `SELECT`, `INSERT` and other queries which are using the table to finish. The table will be removed when it's not in use.

Reviewer: ```suggestion
You can enable synchronous mode using the `SYNC` modifier. To do so, use [`database_atomic_wait_for_drop_and_detach_synchronously`](../../operations/settings/settings.md#database_atomic_wait_for_drop_and_detach_synchronously) setting. In this case, the `DROP TABLE` waits for any running `SELECT`, `INSERT` and other queries using the table to complete. The table is removed only when it's not in use.
```

### EXCHANGE TABLES/DICTIONARIES {#exchange-tables}

> The [`EXCHANGE`](../../sql-reference/statements/exchange.md) query swaps tables or dictionaries atomically. For instance, instead of this non-atomic operation:

Reviewer: ```suggestion
The [`EXCHANGE`](../../sql-reference/statements/exchange.md) query swaps tables or dictionaries atomically. For example, instead of this non-atomic operation:
```

```sql title="Non-atomic"
RENAME TABLE new_table TO tmp, old_table TO new_table, tmp TO old_table;
```
you can use an atomic one:

```sql title="Atomic"
EXCHANGE TABLES new_table AND old_table;
```

### ReplicatedMergeTree in Atomic Database {#replicatedmergetree-in-atomic-database}

> For [`ReplicatedMergeTree`](../table-engines/mergetree-family/replication.md#table_engines-replication) tables, it is recommended not to specify the engine parameters for the path in ZooKeeper and the replica name. In this case, configuration parameters [`default_replica_path`](../../operations/server-configuration-parameters/settings.md#default_replica_path) and [`default_replica_name`](../../operations/server-configuration-parameters/settings.md#default_replica_name) will be used. If you want to specify engine parameters explicitly, it is recommended to use `{uuid}` macros. This is useful so that unique paths are automatically generated for each table in ZooKeeper.

Reviewer: ```suggestion
For [`ReplicatedMergeTree`](../table-engines/mergetree-family/replication.md#table_engines-replication) tables, it is recommended not to specify the engine parameters for the path in ZooKeeper and the replica name. In this case, the configuration parameters [`default_replica_path`](../../operations/server-configuration-parameters/settings.md#default_replica_path) and [`default_replica_name`](../../operations/server-configuration-parameters/settings.md#default_replica_name) will be used. If you want to specify engine parameters explicitly, it is recommended to use the `{uuid}` macros. This ensures that unique paths are automatically generated for each table in ZooKeeper.
```

## See Also

> 

Reviewer: The [system.databases](../../operations/system-tables/databases.md) system table.
